### ISSUE TITLE
[MCS-DE-GO] Syntax Validation does not catch all .LU syntax errors

### ISSUE THREAD

**Issue by goergenj** [microsoft/BotFramework-Composer#3675]
On our implementation we found that not all .LU-file syntax errors are being recognized.

We had this (erroneous) entry:

#GetConfirmation
- yes of course
- I'd like that yes
- I'd love to
- no thank you
- not today thanks
- no rather not

```
# GetConfirmation
- **_{confirmation = yes}_** of course
- I'd like that **_{confirmation = yes}_**

@ list confirmation _**(no "=")**_
	- yes :
	     - of course; sure**_;_**
    - no :
	     - no way; i don't
```

This is the correct syntax:

```
# GetConfirmation
- yes of course
- I'd like that yes

@ list confirmation =
	- yes :
	     - of course; sure
    - no :
	     - no way; i don't
```

Please check, if the .LU-file Syntaxchecker can be updated/improved.

@joaollq @sgellock @benbrown

**Comment by feich-ms** [comment]
@vishwacsena I tested above 'syntax-invalid' lu content, they didn't hit any errors or warnings in lu parser. This is expected behavior in parser since our rules are designed as loose as possible so that we can handle errors or make corrections more friendly when constructing luis json. These syntax errors seems having no impact on the intent or entity definition parsing. Do you think we should handle these no-impact syntax errors when constructing luis json?

**Comment by vishwacsena** [comment]
@feich-ms agreed. @goergenj the parser is pretty lenient in cases where the inconsistencies are inconsequential. You should have the exact same output LUIS app in both these cases. Can you clarify why you believe these should be reported as errors by the syntax checker?

**Comment by goergenj** [comment]
@vishwacsena Have you tested deploying a LUIS app with the errorneous syntax? This is where our process failed which is why we raised this as an issue. ANything that cannot be properly deployed and will cause a deployment to be inconsistent or stalling should be flagged by the syntax checker, of possible.

**Comment by goergenj** [comment]
@benbrown Any update on this after Vishwac has left?

**Comment by goergenj** [comment]
@cwhitten Can you provide any insights to this? Ben is on vacation and I believe vishvac has left?

**Comment by cwhitten** [comment]
Adding this into the backlog for further discussion.

**Comment by cwhitten** [comment]
@feich-ms, putting this on you for R12
